### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Dongmo12**

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Board
+    attr_accessor :post
+     def initialize()
+        @post = [ 1,2,3,4,5,6,7,8,9 ]
+     end
+    def display
```
Comment (AdnanAfsari) [inline lib/board.rb]: User interaction commands like puts and get.chomp should be in one executable file main.rb.
Game logic should be kept in classes (no “puts” and “get.chomp” there!)

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Player
+    attr_accessor :name , :playSymbol
```
Comment (AdnanAfsari) [inline lib/player.rb]: Use attr_readers/attr_writers instead of atrr_acessors.

File: lib/game_logic.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    end
+
+    def board_full?
+      return false if @board.post.any?{|x| x.is_a?(Integer)} 
+
+      true
+    end
+
+  
+    def game_play
```
Comment (AdnanAfsari) [inline lib/game_logic.rb]: The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Studentes tends to create long methods that are doing multiple steps of the game process.

so please refactor it and do not forget to move all the User interactions from game_logic to main.rb

**Review by AdnanAfsari — CHANGES_REQUESTED** [review]
Hello @Dongmo12 @ibrolive ✋ 

Great Work guys but you have to make changes and refactor your code a little bit. please check my comments for that.
Friendly reminder: You should use Github/Git Flow in this project. However, changing it now would take too much time.
Please make sure you use Github/Git Flow in the next project. You will find instructions in the Patwright lesson description.

Submit for another review after making those changes 😄 

Happy Coding

**Review by SunnyAsar — CHANGES_REQUESTED** [review]
Hello @Dongmo12  and @ibrolive 👋 
Nice cleaning up the game logic as discussed in last review.
- **your app works nicely too!**
But:
- we have major linting problems in the code base. please install a linter locally  so you can catch such errors before pushing the code. also install a linter here on github. try ruboCop.
- let's try to follow git flow best practices and not make pull requests to master. [github flow](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/)
This should be used for your next project.

please make the upadates and request another review. cheers.

File: .stickler.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+linters:
+  rubocop:
+    display_cop_names: true
```
Comment (bolah2009) [inline .stickler.yml:3]: #### `REQUIRED`

Let's add a corresponding `.rubocop.yml` config file to the root directory too.

You can find the `.rubocop.yml` file [**here**](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml).

`#.rubocop.yml`
```yml
AllCops:
  Exclude:
    - "bin/*"
    - "db/*"
    - "config/*"
    - "Guardfile"
    - "Rakefile"
  DisplayCopNames: true

Metrics/LineLength:
  Max: 120
Metrics/MethodLength:
  Include: ["app/controllers/*"]
  Max: 20
Metrics/AbcSize:
  Include: ["app/controllers/*"]
  Max: 30
Metrics/ClassLength:
  Max: 150
Metrics/BlockLength:
  ExcludedMethods: ['describe']

Style/Documentation:
  Enabled: false
Style/ClassAndModuleChildren:
  Enabled: false

Layout/AlignHash:
  EnforcedColonStyle: key
Layout/ExtraSpacing:
  AllowForAlignment: false
Layout/MultilineMethodCallIndentation:
  Enabled: true
  EnforcedStyle: indented
```

File: bin/main.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+user_two = Player.new('user_two', 'T')
+
+def receive_player_one_input
+  display_message 'enter username for player_one, the first letter of this name
+    will be your player character'
+  @player_one = gets.chomp.to_s.capitalize
+  @userid1 = @player_one[0].upcase
+end
+
+def receive_player_two_input
```
Comment (bolah2009) [inline bin/main.rb]: ### `REQUIRED`

Why do we have the different method (`#receive_player_one_input` and `#receive_player_two_input`) doing exactly the same thing...? :thinking: 

> Read about [DRY rule](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and use it.

**Comment by Oghenebrume50** [comment]
Reviewing.....

**Review by bellom — CHANGES_REQUESTED** [review]
Hi Team, 
You have done well,according to previous review just the linter to be install left, below link will help.
kindly add the linter and fix all errors if need too, checkout this [link](https://github.com/microverseinc/linters-config#troubleshooting) for help.

You should have below screenshot on your github PR page;
image

_Kindly re-submit for review after adding the linter :slightly_smiling_face: Thumps -up await you_

File: bin/main.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+board = Board.new
+user_one = Player.new('user_one', 'O')
+user_two = Player.new('user_two', 'T')
+
+def receive_player_one_input
+  display_message 'enter username for player_one, the first letter of this name
+    will be your player character'
+  @player_one = gets.chomp.to_s.capitalize
+  @userid1 = @player_one[0].upcase
```
Comment (bolah2009) [inline bin/main.rb]: ### `REQUIRED`

:bug: This is a bug! Imagine two players with the same first letter in their name played the game. The game will see the two different players input as one. Illustration below:

```ruby
$ ruby bin/main.rb 
welcome to the tic-tac-toe game 
         ___________________ 
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     1 | 2 | 3     |
        |    ---|---|---    |
        |     4 | 5 | 6     |
        |    ---|---|---    |
        |     7 | 8 | 9     |
        |    ---|---|---    |
        |___________________|
enter username for player_one, the first letter of this name
    will be your player character
Armand
enter username for player_two, the first letter of this name
    will be your player character
Adnan
         ___________________ 
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     1 | 2 | 3     |
        |    ---|---|---    |
        |     4 | 5 | 6     |
        |    ---|---|---    |
        |     7 | 8 | 9     |
        |    ---|---|---    |
        |___________________|
Armand, select an option from 1...9
1
         ___________________ 
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     A | 2 | 3     |
        |    ---|---|---    |
        |     4 | 5 | 6     |
        |    ---|---|---    |
        |     7 | 8 | 9     |
        |    ---|---|---    |
        |___________________|
Adnan, select an option from 1...9
2
         ___________________ 
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     A | A | 3     |
        |    ---|---|---    |
        |     4 | 5 | 6     |
        |    ---|---|---    |
        |     7 | 8 | 9     |
        |    ---|---|---    |
        |___________________|
Armand, select an option from 1...9
3
GAME OVER!
Armand Wins!
         ___________________ 
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     A | A | A     |
        |    ---|---|---    |
        |     4 | 5 | 6     |
        |    ---|---|---    |
        |     7 | 8 | 9     |
        |    ---|---|---    |
        |___________________|
````

### Armand(@Dongmo12 ) didn't actually win, he played only twice... :disappointed:

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Awesome Job Team :ok_hand: 

Love the game flow and descriptive readme... :+1: 

Kindly check out the comments below and submit another code review request. **Please note that this is not Milestone 1.**

[Spoiler Alert :bellhop_bell: ] There's a BUG :bug: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Comment by Oghenebrume50** [comment]
Reviewing.......

**Comment by maelfosso** [comment]
@Oghenebrume50 Problem !!!
I'm also reviewing it

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello coders,

Nice improvements !!!
You did a really nice job !!!
The bug was corrected !! It's a pleasure the play the game.

The only thing to do is to fix the stickler-ci  errors.

Kindly fix it and re-submit a code review request

**Review by bolah2009 — APPROVED** [review]
# It's a :+1:  from me, team :clap: 

I see bugs have been crushed and linter has been fixed... :ok_hand: 

```
        |                   |
        |-----TIC-TAC-TOE---|
        |___________________|
        |     X | O | X     |
        |    ---|---|---    |
        |     X | O | O     |
        |    ---|---|---    |
        |     O | X | X     |
        |    ---|---|---    |
        |___________________|
board full!
STALEMATE!
```
### :wink: 
## Nice Job and good game flow...

Good luck with your next project. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Pull request merged**